data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/CsvLogstashPluginAttributesMapper.java

```diff
@@ -0,0 +1,113 @@
+                LOGSTASH_AUTOGENERATE_COLUMN_NAMES_ATTRIBUTE_NAME);
+
+        if (logstashAttributeExistsAndIsTrue(autogenerateColumnNamesAttribute)) {
+            pluginSettings.put(
+                    DATA_PREPPER_COLUMN_NAMES,
+                    new ArrayList<String>()
+            );
+        }
+
+        final Optional<LogstashAttribute> autodetectColumnNamesAttribute = findLogstashAttribute(logstashAttributes,
+                LOGSTASH_AUTODETECT_COLUMN_NAMES_ATTRIBUTE_NAME);
+
+        if (logstashAttributeExistsAndIsTrue(autodetectColumnNamesAttribute)) {
+            throw new LogstashConfigurationException(LOGSTASH_AUTODETECT_COLUMN_NAMES_EXCEPTION_MESSAGE);
+        }
+
+        final Optional<LogstashAttribute> skipEmptyRowsAttribute = findLogstashAttribute(logstashAttributes,
+                LOGSTASH_SKIP_EMPTY_ROWS_ATTRIBUTE_NAME);
+
+        if (skipEmptyRowsAttribute.isPresent()) {
+            throw new LogstashConfigurationException(LOGSTASH_SKIP_EMPTY_ROWS_EXCEPTION_MESSAGE);
+        }
+
+        final Optional<LogstashAttribute> skipEmptyColumnsAttribute = findLogstashAttribute(logstashAttributes,
+                LOGSTASH_SKIP_EMPTY_COLUMNS_ATTRIBUTE_NAME);
+        if (skipEmptyColumnsAttribute.isPresent()) {
+            throw new LogstashConfigurationException(LOGSTASH_SKIP_EMPTY_COLUMNS_EXCEPTION_MESSAGE);
+        }
+        final Optional<LogstashAttribute> skipHeaderAttribute = findLogstashAttribute(logstashAttributes,
+                LOGSTASH_SKIP_HEADER_ATTRIBUTE_NAME);
+        if (skipHeaderAttribute.isPresent()) {
+            throw new LogstashConfigurationException(LOGSTASH_SKIP_HEADER_EXCEPTION_MESSAGE);
+        }
+        final Optional<LogstashAttribute> convertAttribute = findLogstashAttribute(logstashAttributes,
+                LOGSTASH_CONVERT_ATTRIBUTE_NAME);
+        if (convertAttribute.isPresent()) {
+            throw new LogstashConfigurationException(LOGSTASH_CONVERT_EXCEPTION_MESSAGE);
+        }
+    }
+
+    private Optional<LogstashAttribute> findLogstashAttribute(final List<LogstashAttribute> logstashAttributes,
+                                                              final String logstashAttributeName) {
+        return logstashAttributes.stream()
+                .filter(logstashAttribute -> logstashAttribute.getAttributeName().equals(logstashAttributeName))
+                .findFirst();
+    }
+
+    private boolean logstashAttributeExistsAndIsTrue(final Optional<LogstashAttribute> optionalLogstashAttribute) {
+        return optionalLogstashAttribute.isPresent() && optionalLogstashAttribute.get().getAttributeValue().getValue().equals(true);
+    }
+
+    @Override
+    protected HashSet<String> getCustomMappedAttributeNames() {
+        final int numberOfCustomAttributeNames = 6;
+        final HashSet<String> names = new HashSet<>(numberOfCustomAttributeNames);
+        names.add(DATA_PREPPER_COLUMN_NAMES_SOURCE_KEY);
```

Comment:
Source key is String in both Logstash and Data Prepper, this attribute doesn't require custom mapper.